my_pyscf/dmet/_dmet.py

```diff
@@ -0,0 +1,572 @@
+import numpy as np
+import scipy
+from pyscf import gto, ao2mo, lib, scf
+from mrh.my_pyscf.dmet.localization import Localization
+from mrh.my_pyscf.dmet.fragmentation import Fragmentation
+from mrh.my_pyscf.dmet.basistransformation import BasisTransform
+
+# Author: Bhavnesh Jangid <jangidbhavnesh@uchicago.edu>
+
+# This class will give you the mean-field object, which can be used to run the MR and post-HF
+# calculations.
+
+get_basis_transform = BasisTransform._get_basis_transformed
+
+def is_close_to_integer(num, tolerance=1e-6):
+    warning_msg = f'SVD for this RDM has some problem {(np.abs(num - np.round(num)) < tolerance)}'
+    assert (np.abs(num - np.round(num)) < tolerance), warning_msg 
+
+def check_orthoganality(mo_coeff, ORTH_THRESH=1e-7):
+    norb = mo_coeff.shape[1]
+    deviation = np.linalg.norm(np.dot(mo_coeff.T, mo_coeff) - np.identity(norb))
+    assert deviation < ORTH_THRESH, "Embedded orbitals are not orthonormal."
+
+def perform_schmidt_decomposition_type1(rdm, nfragorb, nlo, bath_tol=1e-5):
+    '''
+    This way of schmidt decomposition corresponds to previous implementation
+    where one will consider all the singly occupied orbital in the bath, whether they
+    are in impurity space or not.
+
+    Not a wise way of doing it, still having it to make sure we have functionalies of previous
+    implementations.
+
+    Args:
+        rdm: np.ndarray (2, n, n); 
+             n=number of enviroment orbitals
+        nfragorb: int
+            no of fragment orb
+        nlo: int
+            no of localized orb, basically this will be same as that of total number of atomic orbital.
+        bath_tol: float
+            threshold to keep an orbital in the embedding space or through it out.
+    return
+        u_selected: np.array (nlo, neo)
+            unitary matrix which will transform the localized orbitals to the bath orbitals 
+        u_core: np.array(nlo, ncore)
+            unitary matrix which will transform the localized orbitals to the core orbitals 
+    '''
+    # Selecting all the orbitals for which have occupancy anywhere 
+    # between bath tol and doubly occupied.
+    u, bathocc = np.linalg.svd(np.sum(rdm, axis=0))[:2]
+    idx_emb = np.where((bathocc > bath_tol) & (bathocc < 2))[0] 
```

Comment:
use `(2-bathocc) > bath_tol` here, because rounding errors and finite-precision arithmetic might lead to bathocc = 1.99999999999999999999